docs/large_project_structure.md

```diff
@@ -0,0 +1,107 @@
+If you have multiple a large project with maybe multiple `streams` we recommend the following project structure:
+
+```
+├── my-project
+│   ├── my_project
+│   │   ├── __init__.py
+│   │   ├── app.py
+│   │   ├── resources.py
+│   │   └── streams.py
+│   │── pyproject.toml
+│   │── README.md
+```
+
+- The file `my_project/resouces.py` contains the creation of the `StreamEngine`
+- The file `my_project/app.py` contains the entrypoint of your program
+- The file `my_project/streams.py` contains all the `Streams`
+
+A full project example ready to use can be found [here](https://github.com/kpn/kstreams/tree/master/examples/recommended-worker-app)
+
+!!! note
+    This is just a recommendation, there are many ways to structure your project
+
+## Resources
+
+This python module contains any global resource that will be used later in the application, for example `DB connections` or the `StreamEngine`. Typically we will have the following:
```

Comment:
For me it is not entirely clear why you would want to have a separate file for the resources. Especially since there is so little config going on, is that really necessary?

If it is, could you provide one or more extra examples? Now I don't really see the added benefit of having a separate resources file. (Or maybe some extra config lines that you could put here other than just the .create_engine)

Reply:
It's also a way to avoid circular imports in python. A lot of projects require global instantiation, and then let's say you create the `stream_engine` in the `app.py`, then you want to do a `stream_engine.send()` from `streams.py` function, you have a problem.